File: Interview_Prep_App/native_java/retrofit/src/main/java/com/dreams/interviewprepapp/ui/questions/DialogFragment.java
package com.dreams.interviewprepapp.ui.questions;

import android.content.DialogInterface;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;
import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import com.dreams.interviewprepapp.repositories.models.response.Answer;

import java.util.concurrent.Callable;

public class DialogFragment extends androidx.fragment.app.DialogFragment {
    private final Answer answer;
    Callable<String> func;

    private TextView shortAnswerTv, detailTv;


    public DialogFragment(Answer answer, Callable<String> func) {
        this.answer = answer;
        this.func = func;
    }

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        super.onCreateView(inflater, container, savedInstanceState);
        View view = inflater.inflate(com.dreams.interviewprepapp.R.layout.fragment_dialog, container, false);

        shortAnswerTv = view.findViewById(com.dreams.interviewprepapp.R.id.short_ans_tv);
        detailTv = view.findViewById(com.dreams.interviewprepapp.R.id.details_tv);
        shortAnswerTv.setText("\"" + answer.getAnswer() + "\"");
        detailTv.setText(answer.getDetails());
        return view;
    }

    @Override
    public void onDismiss(@NonNull DialogInterface dialog) {
        super.onDismiss(dialog);

        try {
            func.call();
        } catch(Exception e) {
            e.printStackTrace();
        }
    }


}

File: Interview_Prep_App/native_kotlin/base/src/main/java/com/nudennie/interviewprepapp/ui/questions/DialogFragment.kt
package com.nudennie.interviewprepapp.ui.questions

import android.content.DialogInterface
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.fragment.app.DialogFragment
import com.nudennie.interviewprepapp.R
import com.nudennie.interviewprepapp.repositories.models.response.Answer

class DialogFragment(private val answer: Answer, private val func: () -> Unit) : DialogFragment() {
    // Declaring variables
    private lateinit var shortAnswerTv: TextView
    private lateinit var detailTv: TextView

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        super.onCreateView(inflater, container, savedInstanceState)
        // Inflating view
        val view: View = inflater.inflate(R.layout.fragment_dialog, container, false)
        // Connecting short answer and deatils to their views
        shortAnswerTv = view.findViewById(R.id.short_ans_tv)
        detailTv = view.findViewById(R.id.details_tv)

        // Assigning the text to be displayed in short answer and details
        shortAnswerTv.text = "\"" + answer.answer + "\""
        detailTv.text = answer.details
        return view
    }

    // Function on what to do when a dialog is dismissed
    override fun onDismiss(dialog: DialogInterface) {
        super.onDismiss(dialog)
        // Try catch block to catch exceptions
        try {
            func()
        } catch (e: Exception) {
            // Printing an exception if it is there
            e.printStackTrace()
        }
    }
}
